pysages/backends/common.py

```diff
@@ -0,0 +1,22 @@
+# SPDX-License-Identifier: MIT
+# Copyright (c) 2020-2021: PySAGES contributors
+# See LICENSE.md and CONTRIBUTORS.md at https://github.com/SSAGESLabs/PySAGES
+
+
+def restore(view, restore_vm, snapshot, prev_snapshot):
+    # Create a mutable view of the jax arrays
+    positions = view(snapshot.positions)
+    forces = view(snapshot.forces)
+    ids = view(snapshot.ids)
+    # Overwrite the data
+    positions[:] = view(prev_snapshot.positions)
+    forces[:] = view(prev_snapshot.forces)
+    ids[:] = view(prev_snapshot.ids)
+    # Special handling for velocities and masses
+    restore_vm(snapshot, prev_snapshot)
```

Comment:
I don't understand this section complete.

I understand, that it makes sense to not restore masses and velocities all the time. (some methods/applications don't need it.)
But the restore function, doesn't offer no restore as an option. Am I missing something here?

Reply:
Well, technically you can pass a `restore_vm` here that effectively doesn't do anything, e.g. `lambda s, ps: None`. The method `restore_vm` defined here is a convenience fallback, but you're right in that there is no way for the user to specify that.

Reply:
OK, now I see how the user can easily exchange the method if needed.
And to allow passing a function gives greater flexibility. I like the method.
And passing a NOP lambda should definitely be fine.
When we go and write the documentation, we should probably mention it there.